nixos/doc/manual/development/unit-handling.section.md

```diff
@@ -48,10 +48,19 @@ checks:
     If the flag is enabled, the unit is **stop**ped and then **start**ed. If
     not, the unit is **restart**ed. The goal of the flag is to make sure that
     the new unit never runs in the old environment which is still in place
-    before the activation script is run.
+    before the activation script is run. This behavior is different when the
+    service is socket-activated, as outlined in the following steps.
 
   - The last thing that is taken into account is whether the unit is a service
-    and socket-activated. Due to a bug, this is currently only done when
-    `X-StopIfChanged` is set. If the unit is socket-activated, the socket is
-    stopped and started, and the service is stopped and to be started by socket
-    activation.
+    and socket-activated. If `X-StopIfChanged` is **not** set, the service
+    is **restart**ed with the rest.
+
+  - If the socket-activated service does have `X-StopIfChanged` set (the
+    default), the behavior depends on whether `ExecStop=` is set. If set, the
+    `.socket` unit and the `.service` unit are **stop**ped and the `.socket`
+    unit is **start**ed. This is required to prevent races. If `ExecStop=` is
+    not set, there is no need to touch the `.socket` unit. Instead, the
```

Comment:
This means that despite `stopIfChanged` being enabled, the old service will run until _after_ `etc` has been switched.
I don't think that's acceptable.